pysteps/feature/tstorm.py

```diff
@@ -0,0 +1,224 @@
+    labels, n_groups = ndi.label(max_image)
+    for n in range(1, n_groups + 1):
+        indx, indy = np.where(labels == n)
+        if len(indx) > 3:
+            max_image[indx[0], indy[0]] = 2
+    filt_image[max_image == 2] = maxref + 1
+    binary = np.zeros(filt_image.shape)
+    binary[filt_image > 0] = 1
+    labels, n_groups = ndi.label(binary)
+    for n in range(1, n_groups + 1):
+        ind = np.where(labels == n)
+        size = len(ind[0])
+        maxval = np.nanmax(input_image[ind])
+        if size < minsize:
+            binary[labels == n] = 0
+            labels[labels == n] = 0
+        if maxval < minmax:
+            binary[labels == n] = 0
+            labels[labels == n] = 0
+    filt_image = filt_image * binary
+    if mindis % 2 == 0:
+        elem = mindis - 1
+    else:
+        elem = mindis
+    struct = np.ones([elem, elem])
+    if np.nanmax(filt_image.flatten()) < minref:
+        maxima = np.zeros(filt_image.shape)
+    else:
+        maxima = skim.h_maxima(filt_image, h=mindiff, selem=struct)
+    loc_max = np.where(maxima > 0)
+
+    loc_max = longdistance(loc_max, mindis)
+    i_cell = labels[loc_max]
+    n_cell = np.unique(labels)[1:]
+    for n in n_cell:
+        if n not in i_cell:
+            binary[labels == n] = 0
+            labels[labels == n] = 0
+
+    maxima_dis = np.zeros(maxima.shape)
+    maxima_dis[loc_max] = 1
+
+    areas = belonging(input_image, np.nanmin(input_image.flatten()), maxima_dis)
+
+    cells_id, labels = get_profile(areas, binary, input_image, loc_max, time, minref)
+
+    return cells_id, labels
+
+
+def belonging(ref, minval, maxima):
+    """
+    This function segments the entire 2-D array into areas belonging to each identified
+    maximum according to a watershed algorithm.
+    """
+    ref_t = np.zeros(ref.shape)
+    ref_t[:] = minval
+    ref_t[ref > minval] = ref[ref > minval]
+    markers = ndi.label(maxima)[0]
+    areas = skis.watershed(-ref_t, markers=markers)
+    # lines=skis.watershed(-ref_t, markers=markers, watershed_line=True)
+
+    return areas
+
+
+def longdistance(loc_max, mindis):
+    """
+    This function computes the distance between all maxima and rejects maxima that are
+    less than a minimum distance apart.
+    """
+    x_max = loc_max[1]
+    y_max = loc_max[0]
+    n = 0
+    while n < len(y_max):
+        disx = x_max[n] - x_max
+        disy = y_max[n] - y_max
+        dis = np.sqrt(disx * disx + disy * disy)
+        close = np.where(dis < mindis)[0]
+        close = np.delete(close, np.where(close <= n))
+        if len(close) > 0:
+            x_max = np.delete(x_max, close)
+            y_max = np.delete(y_max, close)
+        n += 1
+
+    new_max = y_max, x_max
+
+    return new_max
+
+
+def get_profile(areas, binary, ref, loc_max, time, minref):
+    """
+    This function returns the identified cells in a dataframe including their x,y
+    locations, location of their maxima, maximum reflectivity and contours.
```

Comment:
Here `x` and `y` are in unit of number of pixels from the field origin, correct?
This might create some confusion later on when, for example, one would try to plot the contours.

Should we instead work in geographical units? All pysteps data must come on a regular grid and the metadata dictionary will include the definition of the geographical projection.

Reply:
At least partially addressed in f0ebdb4